Add documentation of requirements in the underlining databases of Sca… (#158)
scalar-labs/scalardb: docs/requirements.md

For the second, Scalar DB does not work on some Cassandra compatible databases such as [Amazon Keyspaces](https://aws.amazon.com/keyspaces/) that don't support LWT since `ConsensusCommit` transaction manager relies on linearizable operations of underlining databases to make transactions serializable. 

If the above requirements are met, storage operations with `LINEARIZABLE` can provide linearizablity and transaction operations with `SERIALIZABLE` can provide strict serializability.

## JDBC databases

In Scalar DB on JDBC databases, you can't choose a consistency level (`LINEARIZABLE`, `SEQUENTIAL` or `EVENTUAL`) in your code with the `Operation.withConsistency()` method, and the consistency level depends on the setup of your JDBC database.

> For example, when you have asynchronous read replicas in your setup and perform read operations against them, the consistency level will be `EVENTUAL`, because you can read stale data from the read replicas.

Reviewer: ```suggestion
For example, when you have asynchronous read replicas in your setup and perform read operations against them, the consistency will be eventual because you can read stale data from the read replicas.
```
Suggested because the original description sounds like the consistency level is set to `EVENTUAL`.

> Or when you perform all operations against a single master instance, the consistency level will be `LINEARIZABLE`.

Reviewer: ```suggestion
On the other hand, when you perform all operations against a single master instance, the consistency will be linearizable.
```

> We recommend performing all operations/transactions against a single master instance (`LINEARIZABLE`), because reading stale data is very confusing, which causes mistakes easily in your application.

Reviewer: ```suggestion
We recommend performing all operations/transactions against a single master instance (so that you can achieve linearizable) if you want to avoid caring consistency issues in your applications. Note that you can still use a read replica as a backup and standby even if you follow the recommendation.
```

> If you strongly want to use the read replica feature in your setup, you can use the setup where you perform read-write transactions against the master instance and read-only transactions against read replicas.

Reviewer: ```suggestion
If you strongly want to use read replicas in your setup, you can configure your application to perform read-write-mixed transactions against a master instance and read-only transactions against read replicas.
```

> You lose `LINEARIZABLE` consistency level in this setup, but it's still `SERIALIZABLE`.

Reviewer: ```suggestion
The resulting schedule would not be strict serializable (linearizable and serializable) but would be serializable.
```

> If you strongly want to use read replicas in your setup, you can configure your application to perform read-write-mixed transactions against a master instance and read-only transactions against read replicas.

Reviewer: ```suggestion
If you strongly want to perform operations/transactions on read replicas, you can configure your application to perform read-write-mixed transactions against a master instance and read-only transactions against read replicas.
```
Made suggestions for consistency since `use` was a bit ambiguous.
